### PR TITLE
USHIFT-1419: Openvswitch user group workaround

### PR DESCRIPTION
**Pull request by zshi-redhat**

### CODE REVIEW

File: packaging/systemd/microshift-ovsdb-server.conf [hunk 1]
```diff
@@ -1,2 +1,3 @@
 [Service]
 CPUAffinity=0
+ExecStartPre=-/bin/sh -c 'if [ -f /usr/lib/sysusers.d/openvswitch-hugetlbfs.conf ]; then usermod -a -G hugetlbfs openvswitch fi'
```
[the remark this answers is not in the document]
Comment (zshi-redhat) [reply packaging/systemd/microshift-ovsdb-server.conf:3]: openvswitch.service is a dummy systemd service that depends on the other two services: ovs-vswitchd and ovsdb-server.
ovs-vswtichd.service is configured to be `After=ovsdb-server.service`. I was thinking to see if the ExecStartPre in ovsdb-server would be enough as a workaround (if it passes CI). I will add a new systemd unit if it doesn't.

[the remark this answers is not in the document]
Comment (zshi-redhat) [reply packaging/systemd/microshift-ovsdb-server.conf:3]: it seems the cluster is up with openvswitch3.1-3.1.0-33 installed in the CI run: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_microshift/2035/pull-ci-openshift-microshift-main-microshift-e2e-arm/1678939779464433664/artifacts/microshift-e2e-arm/openshift-microshift-infra-rpm-install-aws/build-log.txt

[the remark this answers is not in the document]
Comment (zshi-redhat) [reply packaging/systemd/microshift-ovsdb-server.conf:3]: @dhellmann  I can make this workaround in a separate systemd unit that starts before openvswitch.service if you prefer to, I think the separate systemd unit will a cleaner implementation than the current one.

**Comment by dhellmann** [comment]
The issue with `ostree ref` should be fixed by #2051, but this PR may need to be rebased to pick it up.

**Comment by dhellmann** [comment]
It looks like the metal test job failed with an issue that I think we've fixed in main. You might need to rebase to get the tests to pass.

File: packaging/systemd/microshift-ovsdb-server.conf (hunk 1 above)
Comment (dhellmann) [inline packaging/systemd/microshift-ovsdb-server.conf:3]:
```suggestion
ExecStartPre=-/bin/sh -c 'if [ -f /usr/lib/sysusers.d/openvswitch-hugetlbfs.conf ]; then usermod -a -G hugetlbfs openvswitch; fi'
```

The command is missing a semicolon, so it's failing when the unit starts.

**Comment by dhellmann** [comment]
I think this problem is tied to ostree making the group database read-only. Logged in to an ostree system, the command `usermod -a -G hugetlbfs openvswitch` has no effect at all.

```
# getent group hugetlbfs
hugetlbfs:x:991:
# usermod -a -G hugetlbfs openvswitch
# getent group hugetlbfs
hugetlbfs:x:991:
```

We're going to need to find a different way to create the user and group entry, maybe from the RPM pre script, or to configure openvswitch to use a user and group that already exist on the host.

What is creating the openvswitch user?

**Comment by dhellmann** [comment]
We believe the issue with the CI jobs on this PR are fixed by https://github.com/openshift/microshift/pull/2083, but this PR will need to be rebased to get the fix.

**Comment by dhellmann** [comment]
/hold

Let me link this to a jira ticket

**Comment by zshi-redhat** [comment]
> /cherry-pick release-4.13

Just a note:

@ggiguash found an issue in the rpm spec - the date doesn't match the week day.  He is going to fix it, which probably shall be cherry-picked along with this PR.

```
# Building RPM packages
warning: bogus date in %changelog: Tue Jul  21 2023 Zenghui Shi <zshi@redhat.com> 4.14.0
```

**Comment by dhellmann** [comment]
> > /cherry-pick release-4.13
> 
> Just a note:
> 
> @ggiguash found an issue in the rpm spec - the date doesn't match the week day. He is going to fix it, which probably shall be cherry-picked along with this PR.
> 
> ```
> # Building RPM packages
> warning: bogus date in %changelog: Tue Jul  21 2023 Zenghui Shi <zshi@redhat.com> 4.14.0
> ```

Ack, thanks.

**Pull request merged**